openprovider/apis/params_creator.php

```diff
@@ -177,4 +216,19 @@ private function getMethodParamsTypes(\ReflectionMethod $method): array
 
         return $paramTags;
     }
+
+    /**
+     * @param string $domainName
+     * @return string encoded domain name with punycode
+     */
+    private function idnEncode(string $domainName): string
+    {
+        $idn = new idna_convert();
+        $encodedDomainName = $domainName;
+        if (!preg_match('//u', $domainName)) {
+            $encodedDomainName = utf8_encode($domainName);
+        }
+
+        return $idn->encode($encodedDomainName);
```

Comment:
why don't use idn_to_ascii php function? We use it in core for that, see https://github.com/openprovider/core/blob/master/api/lib/validator/domain/apiDomainFormatValidator.class.php#L84